docs/concepts/fastapi.md

```diff
@@ -0,0 +1,44 @@
+# Integrating Pydantic Models with FastAPI
+
+[FastAPI](https://fastapi.tiangolo.com/) is an enjoyable tool for building web applications in python. One of its core features is the integration with `Pydantic` models, which makes defining and validating data structures straightforward and efficient. In this guide, we will see how creating simple functions that return `Pydantic` models can seamlessly integrate with FastAPI.
+
+## Code Example: Starting a FastAPI App with a POST Request
+
+The following code snippet demonstrates how to start a `FastAPI` app with a POST endpoint. This endpoint accepts and returns data defined by a `Pydantic` model.
+
+```python
+from fastapi import FastAPI
+from pydantic import BaseModel
+
+app = FastAPI()
+
+class User(BaseModel):
+    name: str
+    age: int
+
+@app.post("/endpoint", response_model=User)
+def endpoint_function(data: User) -> User:
```

Comment:
the input should be a different model, we should change this and actually have the function have a openai call using instructor